File: src/main/java/hzt/aoc/day05/Part1BinaryBoarding.java
package hzt.aoc.day05;

import hzt.aoc.day05.model.Seat;
import hzt.collections.ListX;

public class Part1BinaryBoarding extends Day05Challenge {

    public Part1BinaryBoarding() {
        super("part 1", "Find the highest seat ID on a boarding pass in the list. ");
    }

    @Override
    protected int calculateResult(ListX<Seat> seats) {
        return seats.maxOf(seat -> seat.getSeatID(NUMBER_OF_COLUMNS));
    }

    @Override
    protected String getMessage(String result) {
        return String.format("The highest seat ID is: %s%n", result);
    }

}

File: src/main/java/hzt/aoc/day05/Part1BinaryBoarding.kt
package hzt.aoc.day05

import java.util.stream.Collectors
import hzt.aoc.day05.model.Seat

class Part1BinaryBoarding : Day05Challenge("part 1", "Find the highest seat ID on a boarding pass in the list. ") {

    override fun calculateResult(seats: List<Seat>): Int {
        return findHighestSeatID(seats.stream()
                .map { it.getSeatID(NUMBER_OF_COLUMNS) }
                .collect(Collectors.toList()))
    }

    override fun getMessage(result: String): String {
        return String.format("The highest seat ID is: %s%n", result)
    }
}
